modules/Campaigns/PopupCampaignRoi.php

```diff
@@ -41,78 +38,77 @@
  * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
  */
 
+if (!defined('sugarEntry') || !sugarEntry) {
+    die('Not A Valid Entry Point');
+}
 
-
-
-
-
-require_once('include/DetailView/DetailView.php');
-require_once('modules/Campaigns/Charts.php');
-
+require_once __DIR__ . '/../../include/DetailView/DetailView.php';
+require_once __DIR__ . '/../../modules/Campaigns/Charts.php';
 
 global $mod_strings;
 global $app_strings;
 global $app_list_strings;
 global $sugar_version, $sugar_config;
-
 global $theme;
 
+LoggerManager::getLogger()->info('Campaign detail view');
 
+$xtpl = new XTemplate ('modules/Campaigns/PopupCampaignRoi.html');
 
-
-$GLOBALS['log']->info("Campaign detail view");
-
-$xtpl=new XTemplate('modules/Campaigns/PopupCampaignRoi.html');
-
-$campaign_id=$_REQUEST['id'];
-$campaign = BeanFactory::newBean('Campaigns');
-$opp_query1  = "select camp.name, camp.actual_cost,camp.budget,camp.expected_revenue,count(*) opp_count,SUM(opp.amount) as Revenue, SUM(camp.actual_cost) as Investment,
+$campaign_id = $_REQUEST['id'];
+$campaign = new Campaign();
```

Comment:
Hi @Dillon-Brown,

I think there was an MR sometime ago to replace all direct module instantiations with calls to BeanFactory, as that allowed for more extensibility. Is there a reason going back to a direct instantiation? or was it just something that came from the rebase?

Reply:
Nope, definitely don't want to revert the ```BeanFactory``` changes. Must have accidentally taken the wrong side when resolving conflicts here. Pushed.